src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java

```diff
@@ -129,5 +155,42 @@ private void registerWebhook(BitbucketSCMRepository repository) {
             BitbucketWebhook webhook = retryingWebhookHandler.register(repository);
             LOGGER.info("Webhook returned -" + webhook);
         }
+
+        private boolean isWebhookExists(Job<?, ?> project, BitbucketSCM input) {
+            try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+                return jenkinsProvider
+                        .get().getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)
+                        .stream()
+                        .filter(item -> !item.equals(project))
+                        .filter(this::isTriggerEnabled)
+                        .map(item -> asSCMTriggerItem(item))
+                        .filter(Objects::nonNull)
+                        .map(scmItem -> scmItem.getSCMs())
+                        .flatMap(Collection::stream)
+                        .filter(scm -> scm instanceof BitbucketSCM)
+                        .map(scm -> (BitbucketSCM) scm)
+                        .map(BitbucketSCM::getRepositories)
+                        .flatMap(Collection::stream)
+                        .anyMatch(scm -> isExistingWebhookOnRepo(input, scm));
+            }
+        }
+
+        private boolean isTriggerEnabled(ParameterizedJobMixIn.ParameterizedJob job) {
+            return job.getTriggers()
+                    .values()
+                    .stream()
+                    .anyMatch(v -> v instanceof BitbucketWebhookTriggerImpl);
+        }
+
+        private boolean isExistingWebhookOnRepo(BitbucketSCM scm, BitbucketSCMRepository repository) {
+            return scm.getRepositories().stream().allMatch(r -> r.getServerId().equals(repository.getServerId()) &&
```

Comment:
Would this be better having an `equals` and `hashCode` method on `BitbucketSCMRepository`?

Reply:
I like the current one. I don't want the mirrors to be equal.